packages/gatsby-cli/src/reporter/catch-exit-signals.ts

```diff
@@ -5,20 +5,43 @@
 import signalExit from "signal-exit"
 import { getStore } from "./redux"
 import { createPendingActivity } from "./redux/actions"
-import { ActivityStatuses } from "./constants"
+import { ActivityStatuses, ActivityTypes } from "./constants"
 import { reporter } from "./reporter"
 
-const interruptActivities = (): void => {
+const interruptActivities = ({
+  showCancelledPendingActivities = false,
+}: {
+  showCancelledPendingActivities?: boolean
+} = {}): void => {
   const { activities } = getStore().getState().logs
+  const cancelledPendingActivitiesIds: Array<string> = []
   Object.keys(activities).forEach(activityId => {
     const activity = activities[activityId]
     if (
       activity.status === ActivityStatuses.InProgress ||
       activity.status === ActivityStatuses.NotStarted
     ) {
       reporter.completeActivity(activityId, ActivityStatuses.Interrupted)
+      if (activity.type === ActivityTypes.Pending) {
+        // pending activities are hidden from UI, so we construct
+        // list of pending activities to log them to make it easier
+        // to debug any timing issues related to reporting
+        // global status (SET_STATUS)
+        cancelledPendingActivitiesIds.push(`"${activityId}"`)
+      }
     }
   })
+
+  if (
+    showCancelledPendingActivities &&
+    cancelledPendingActivitiesIds.length > 0
```

Comment:
With the wording of "idle process", wouldn't they want to be warned about any activities passing the `InProgress` or `NotStarted` checks?

Reply:
Yeah, I will re-do this part a little - together with what I mentioned in https://github.com/gatsbyjs/gatsby/pull/26332#issuecomment-671465335 which will also need quite a bit more changes